src/ClientRuntime/Microsoft.Rest.ClientRuntime.Tests/CustomClientWithHttpClientTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Rest.ClientRuntime.Tests
+{
+    using Microsoft.Rest.ClientRuntime.Tests.CustomClients;
+    using System.Net.Http;
+    using Xunit;
+
+    public class CustomClientWithHttpClientTests
+    {
+        [Fact]
+        public void InitializeServiceClientWithHttpClient()
+        {
+            HttpClient hc = new HttpClient();
```

Comment:
In one of these methods, we should use the http client once before providing it to the ContosoClient.

We should also assert over all the properties we expect to be set on the HttpClient